<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="James Reineke and Matt Cannon">
<meta name="dcterms.date" content="2023-09-28">

<title>Data Analysis for Fibrosis paper - Loading and initial processing of stroma</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Load-Cluster_files/libs/clipboard/clipboard.min.js"></script>
<script src="Load-Cluster_files/libs/quarto-html/quarto.js"></script>
<script src="Load-Cluster_files/libs/quarto-html/popper.min.js"></script>
<script src="Load-Cluster_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Load-Cluster_files/libs/quarto-html/anchor.min.js"></script>
<link href="Load-Cluster_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Load-Cluster_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Load-Cluster_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Load-Cluster_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Load-Cluster_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sample-summary" id="toc-sample-summary" class="nav-link active" data-scroll-target="#sample-summary">Sample summary</a></li>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link" data-scroll-target="#load-packages">Load packages</a>
  <ul class="collapse">
  <li><a href="#make-up-directory-structure" id="toc-make-up-directory-structure" class="nav-link" data-scroll-target="#make-up-directory-structure">Make up directory structure</a>
  <ul class="collapse">
  <li><a href="#get-cell-type-reference-datasets" id="toc-get-cell-type-reference-datasets" class="nav-link" data-scroll-target="#get-cell-type-reference-datasets">Get cell type reference datasets</a></li>
  </ul></li>
  <li><a href="#functions-to-use-for-cell-annotation" id="toc-functions-to-use-for-cell-annotation" class="nav-link" data-scroll-target="#functions-to-use-for-cell-annotation">Functions to use for cell annotation</a></li>
  </ul></li>
  <li><a href="#load-raw-data-and-suggest-cell-type" id="toc-load-raw-data-and-suggest-cell-type" class="nav-link" data-scroll-target="#load-raw-data-and-suggest-cell-type">Load raw data and suggest cell type</a></li>
  <li><a href="#make-up-seurat-objects-with-subsets-of-the-data" id="toc-make-up-seurat-objects-with-subsets-of-the-data" class="nav-link" data-scroll-target="#make-up-seurat-objects-with-subsets-of-the-data">Make up Seurat objects with subsets of the data</a>
  <ul class="collapse">
  <li><a href="#b6-and-f420" id="toc-b6-and-f420" class="nav-link" data-scroll-target="#b6-and-f420">B6 and F420</a>
  <ul class="collapse">
  <li><a href="#create-the-merged-tumornormal-object" id="toc-create-the-merged-tumornormal-object" class="nav-link" data-scroll-target="#create-the-merged-tumornormal-object">Create the merged tumor/normal object</a></li>
  <li><a href="#inspect-identities-from-singler-calls-and-other-meta" id="toc-inspect-identities-from-singler-calls-and-other-meta" class="nav-link" data-scroll-target="#inspect-identities-from-singler-calls-and-other-meta">Inspect identities from SingleR calls and other meta</a></li>
  <li><a href="#rename-clusters-based-on-the-calls" id="toc-rename-clusters-based-on-the-calls" class="nav-link" data-scroll-target="#rename-clusters-based-on-the-calls">Rename clusters based on the calls</a></li>
  <li><a href="#create-publication-plots-and-save" id="toc-create-publication-plots-and-save" class="nav-link" data-scroll-target="#create-publication-plots-and-save">Create publication plots and save</a></li>
  <li><a href="#create-frequency-plots-of-stromal-and-immune-cells" id="toc-create-frequency-plots-of-stromal-and-immune-cells" class="nav-link" data-scroll-target="#create-frequency-plots-of-stromal-and-immune-cells">Create frequency plots of stromal and immune cells</a></li>
  </ul></li>
  <li><a href="#balbc-and-k7m2" id="toc-balbc-and-k7m2" class="nav-link" data-scroll-target="#balbc-and-k7m2">BALBC and K7M2</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data Analysis for Fibrosis paper - Loading and initial processing of stroma</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>James Reineke and Matt Cannon </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 28, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="sample-summary" class="level1">
<h1>Sample summary</h1>
<ul>
<li>Our data
<ul>
<li>stuff here</li>
</ul></li>
<li>GSE141259
<ul>
<li>Longitudinal single-cell transcriptomics analysis of mouse lung upon bleomycin-induced injury</li>
<li>PMID: 32678092</li>
<li>DropSeq</li>
<li>Has three standard gz files plus metadata</li>
</ul></li>
<li>GSE145031
<ul>
<li>Inflammatory Signals induce AT2 Cell-Derived Damage-Associated Transient Progenitors that Mediate Alveolar Regeneration</li>
<li>PMID: 32750316</li>
<li>10X</li>
<li>Has three standard gz files</li>
</ul></li>
<li>GSE157995
<ul>
<li>Single-cell RNA-sequencing of bleomycin-injured young and old mice</li>
<li>PMID: 35389887</li>
<li>10X</li>
<li>Has csv.gz files with count matrices</li>
<li>Six samples have .h5 files instead</li>
<li>No metadata included and the code is hand annotated</li>
</ul></li>
<li>GSE201698
<ul>
<li>Time-course cellular interactome analysis by single-cell RNA-seq in bleomycin-induced pulmonary fibrosis</li>
<li>not published</li>
<li>BD Rhapsody platform</li>
<li>Has txt.gz files with count matrices</li>
</ul></li>
<li>GSE151374
<ul>
<li>The Anti-Fibrotic Drug Nintedanib Promotes Expansion of Lung Macrophages with a Distinct Transcriptional Repair Program in Bleomycin-Exposed Mice</li>
<li>PMID: 36227799</li>
<li>Think it’s 10x - Not sure - paper not available</li>
<li>Has .h5 files</li>
</ul></li>
</ul>
</section>
<section id="load-packages" class="level1">
<h1>Load packages</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cluster)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter()    masks stats::filter()
✖ dplyr::lag()       masks stats::lag()
✖ lubridate::stamp() masks cowplot::stamp()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'

The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Attaching SeuratObject</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sctransform)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rrrSingleCellUtils)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: nichenetr</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parallel)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set random generator seed to facilitate reproducibility</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">888</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="make-up-directory-structure" class="level2">
<h2 class="anchored" data-anchor-id="make-up-directory-structure">Make up directory structure</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> directoryName <span class="kw">in</span> <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  output <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  output/figures <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  output/rdata <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  output/de</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-d</span> <span class="va">${directoryName}</span> <span class="bu">]</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">then</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${directoryName}</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">fi</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="get-cell-type-reference-datasets" class="level3">
<h3 class="anchored" data-anchor-id="get-cell-type-reference-datasets">Get cell type reference datasets</h3>
<p>GSE151974</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>ref_path <span class="ot">&lt;-</span> <span class="st">"/home/gdrobertslab/lab/GenRef/sc_ref_datasets/mouse"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>mouse_lung_ref <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(<span class="fu">paste0</span>(ref_path, <span class="st">"/GSE151974/mouse_lung_ref.qs"</span>))</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>mouse_immune <span class="ot">&lt;-</span> celldex<span class="sc">::</span><span class="fu">ImmGenData</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>snapshotDate(): 2022-10-31</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This label is not informative</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>mouse_immune <span class="ot">&lt;-</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    mouse_immune[mouse_immune<span class="sc">$</span>label.main <span class="sc">!=</span> <span class="st">"Stromal cells"</span>, ]</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>mouse_rna <span class="ot">&lt;-</span> celldex<span class="sc">::</span><span class="fu">MouseRNAseqData</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>snapshotDate(): 2022-10-31
see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Corrupt Cache: resource path
  See AnnotationHub's TroubleshootingTheCache vignette section on corrupt cache
  cache: /gpfs0/home2/gdrobertslab/rxr014/.cache/R/ExperimentHub
  potential duplicate files: 
    21f52c3334adad_3510
    d25db4b09557f_3510
    21f52c5b15d944_3511
    d25db76317a50_3511
    21f52c6351e0e6_3514
    9383c29bf7021_3514
    21f52c595d46ce_3515
    9383c2b628b1c_3515
Continuing with first found cached file</code></pre>
</div>
</div>
</section>
</section>
<section id="functions-to-use-for-cell-annotation" class="level2">
<h2 class="anchored" data-anchor-id="functions-to-use-for-cell-annotation">Functions to use for cell annotation</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>annot_cells <span class="ot">&lt;-</span> <span class="cf">function</span>(sobject) {</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>    cell_assign <span class="ot">&lt;-</span> SingleR<span class="sc">::</span><span class="fu">SingleR</span>(</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.SingleCellExperiment</span>(sobject),</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">ref =</span> <span class="fu">list</span>(<span class="fu">GetAssayData</span>(mouse_lung_ref), mouse_immune, mouse_rna),</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">list</span>(</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>            mouse_lung_ref<span class="sc">$</span>cell_type,</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>            mouse_immune<span class="sc">$</span>label.main,</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>            mouse_rna<span class="sc">$</span>label.fine),</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">aggr.ref =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>    sobject<span class="sc">$</span>cell_type <span class="ot">&lt;-</span> cell_assign<span class="sc">$</span>labels</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>    sobject<span class="sc">$</span>cell_score <span class="ot">&lt;-</span> cell_assign<span class="sc">$</span>scores <span class="sc">%&gt;%</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">apply</span>(<span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">max</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(sobject)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>crazy_cols <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">seed =</span> <span class="dv">1337</span>) {</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(seed)</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sample</span>(<span class="fu">rainbow</span>(<span class="at">n =</span> <span class="fu">length</span>(<span class="fu">unique</span>(x))))</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="load-raw-data-and-suggest-cell-type" class="level1">
<h1>Load raw data and suggest cell type</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>sample_list <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"misc/sample_cutoffs.txt"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>end_path <span class="ot">&lt;-</span> <span class="st">"/filtered_feature_bc_matrix"</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">grepl</span>(<span class="st">"r1pl"</span>, <span class="fu">Sys.info</span>()[[<span class="st">"nodename"</span>]])) {</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>    data_path <span class="ot">&lt;-</span> <span class="st">"/home/gdrobertslab/lab/Counts/"</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>    path_use <span class="ot">&lt;-</span> <span class="st">"cluster_path"</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>    data_path <span class="ot">&lt;-</span> <span class="st">"/Applications/scRNA_seq raw files/"</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>    path_use <span class="ot">&lt;-</span> <span class="st">"James_path"</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>sobj_list <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(sample_list)),</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">mc.cores =</span> parallelly<span class="sc">::</span><span class="fu">availableCores</span>(),</span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(i) {</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>        obj_name <span class="ot">&lt;-</span> sample_list<span class="sc">$</span>obj_name[i]</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">message</span>(obj_name)</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>        sobj <span class="ot">&lt;-</span> <span class="fu">tenx_load_qc</span>(</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste0</span>(data_path,sample_list[[path_use]][i], end_path),</span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>            <span class="at">min_cells =</span> <span class="dv">3</span>,</span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>            <span class="at">min_features =</span> <span class="dv">200</span>,</span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">violin_plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Add metadata to dataset</span></span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (colname <span class="cf">in</span> <span class="fu">colnames</span>(sample_list)) {</span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>            sobj[[colname]] <span class="ot">&lt;-</span> sample_list[[colname]][i]</span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Add sample name to dataset</span></span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a>        sobj<span class="sc">$</span>sample <span class="ot">&lt;-</span> obj_name</span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>        cutoff_table <span class="ot">&lt;-</span></span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>            <span class="fu">tribble</span>(<span class="sc">~</span><span class="st">"feature"</span>,   <span class="sc">~</span><span class="st">"min_val"</span>,                    <span class="sc">~</span><span class="st">"max_val"</span>,</span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"nCount_RNA"</span>, sample_list<span class="sc">$</span>ncount_rna_min[i], sample_list<span class="sc">$</span>ncount_rna_max[i],</span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"percent.mt"</span>, <span class="dv">0</span>,                             sample_list<span class="sc">$</span>mt_max[i])</span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>        plotted <span class="ot">&lt;-</span> <span class="fu">feature_hist</span>(sobj,</span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a>            <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nCount_RNA"</span>, <span class="st">"percent.mt"</span>),</span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a>            <span class="at">cutoff_table =</span> cutoff_table)</span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggsave</span>(<span class="fu">paste0</span>(<span class="st">"output/figures/feature_hist_"</span>, obj_name, <span class="st">".png"</span>),</span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a>            <span class="at">width =</span> <span class="dv">10</span>,</span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a>            <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb31-44"><a href="#cb31-44" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot =</span> plotted)</span>
<span id="cb31-45"><a href="#cb31-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-46"><a href="#cb31-46" aria-hidden="true" tabindex="-1"></a>        sobj <span class="ot">&lt;-</span> sobj <span class="sc">%&gt;%</span></span>
<span id="cb31-47"><a href="#cb31-47" aria-hidden="true" tabindex="-1"></a>            <span class="fu">subset</span>(nCount_RNA   <span class="sc">&gt;=</span> sample_list<span class="sc">$</span>ncount_rna_min[i] <span class="sc">&amp;</span></span>
<span id="cb31-48"><a href="#cb31-48" aria-hidden="true" tabindex="-1"></a>                nCount_RNA   <span class="sc">&lt;=</span> sample_list<span class="sc">$</span>ncount_rna_max[i] <span class="sc">&amp;</span></span>
<span id="cb31-49"><a href="#cb31-49" aria-hidden="true" tabindex="-1"></a>                percent.mt   <span class="sc">&lt;=</span> sample_list<span class="sc">$</span>mt_max[i]) <span class="sc">%&gt;%</span></span>
<span id="cb31-50"><a href="#cb31-50" aria-hidden="true" tabindex="-1"></a>            <span class="fu">process_seurat</span>()</span>
<span id="cb31-51"><a href="#cb31-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-52"><a href="#cb31-52" aria-hidden="true" tabindex="-1"></a>        sobj <span class="ot">&lt;-</span> <span class="fu">annot_cells</span>(sobj)</span>
<span id="cb31-53"><a href="#cb31-53" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(sobj)</span>
<span id="cb31-54"><a href="#cb31-54" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb31-55"><a href="#cb31-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-56"><a href="#cb31-56" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sobj_list) <span class="ot">&lt;-</span> sample_list<span class="sc">$</span>obj_name</span>
<span id="cb31-57"><a href="#cb31-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-58"><a href="#cb31-58" aria-hidden="true" tabindex="-1"></a>qs<span class="sc">::</span><span class="fu">qsave</span>(sobj_list, <span class="at">file =</span> <span class="st">"output/rdata/sobj_list.qs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="make-up-seurat-objects-with-subsets-of-the-data" class="level1">
<h1>Make up Seurat objects with subsets of the data</h1>
<section id="b6-and-f420" class="level2">
<h2 class="anchored" data-anchor-id="b6-and-f420">B6 and F420</h2>
<section id="create-the-merged-tumornormal-object" class="level3">
<h3 class="anchored" data-anchor-id="create-the-merged-tumornormal-object">Create the merged tumor/normal object</h3>
<p>Process into a shared UMAP space</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>sobj_list <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(<span class="st">"output/rdata/sobj_list.qs"</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>b6_f420_combined <span class="ot">&lt;-</span> <span class="fu">merge</span>(sobj_list[[<span class="st">"C57BL6"</span>]],</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> sobj_list[[<span class="st">"F420"</span>]],</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">add.cell.ids =</span> <span class="fu">c</span>(<span class="st">"C57BL6"</span>, <span class="st">"F420"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">SCTransform</span>(<span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">RunPCA</span>(<span class="at">npcs =</span> <span class="dv">30</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">seed.use =</span> <span class="dv">22</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FindNeighbors</span>(<span class="at">k.param =</span> <span class="dv">30</span>, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FindClusters</span>(<span class="at">resolution  =</span> <span class="fl">0.3</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
This message will be shown once per session</code></pre>
</div>
</div>
</section>
<section id="inspect-identities-from-singler-calls-and-other-meta" class="level3">
<h3 class="anchored" data-anchor-id="inspect-identities-from-singler-calls-and-other-meta">Inspect identities from SingleR calls and other meta</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check identities from SingleR calls and other meta</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(b6_f420_combined, <span class="at">group.by =</span> <span class="st">"seurat_clusters"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_2-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(b6_f420_combined, <span class="at">group.by =</span> <span class="st">"cell_type"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_2-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(b6_f420_combined, <span class="at">features =</span> <span class="st">"cell_score"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_2-3.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(b6_f420_combined, <span class="at">features =</span> <span class="st">"nCount_RNA"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_2-4.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(b6_f420_combined, <span class="at">features =</span> <span class="st">"Col1a1"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_2-5.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(b6_f420_combined, <span class="at">features =</span> <span class="st">"Col1a2"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_2-6.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="rename-clusters-based-on-the-calls" class="level3">
<h3 class="anchored" data-anchor-id="rename-clusters-based-on-the-calls">Rename clusters based on the calls</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">file.exists</span>(<span class="st">"misc/b6_f420_assignments.qs"</span>)) {</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>    b6_f420_combined <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(b6_f420_combined,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>        qs<span class="sc">::</span><span class="fu">qread</span>(<span class="st">"misc/b6_f420_assignments.qs"</span>))</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># This list is manually curated from the SingleR assignments and scores</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If run again, this needs to be checked for accuracy, as clusters may change</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>    b6_f420_combined <span class="ot">&lt;-</span> <span class="fu">RenameIdents</span>(b6_f420_combined,</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">0</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Endo"</span>,</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">4</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">5</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">6</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Endo"</span>,</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">7</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"NK/T"</span>,</span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">8</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Fibroblast"</span>,</span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">9</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"F420"</span>,</span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">10</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"B"</span>,</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">11</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">12</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"NK/T"</span>,</span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">13</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Lower Airway"</span>,</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">14</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Upper Airway"</span>,</span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">15</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mono"</span>,</span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">16</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"F420"</span>,</span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">17</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Upper Airway"</span>,</span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">18</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Granulo"</span>,</span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">19</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"DC"</span>,</span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">20</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"DC"</span>,</span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">21</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Erythro"</span>,</span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">22</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"DC"</span>,</span>
<span id="cb44-31"><a href="#cb44-31" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">23</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Peri"</span>,</span>
<span id="cb44-32"><a href="#cb44-32" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">24</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb44-33"><a href="#cb44-33" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">25</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb44-34"><a href="#cb44-34" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">26</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Adipo"</span>,</span>
<span id="cb44-35"><a href="#cb44-35" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">27</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"F420"</span>,</span>
<span id="cb44-36"><a href="#cb44-36" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">28</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Meso"</span>,</span>
<span id="cb44-37"><a href="#cb44-37" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">29</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Endo"</span>,</span>
<span id="cb44-38"><a href="#cb44-38" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">30</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"SMC"</span>,</span>
<span id="cb44-39"><a href="#cb44-39" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">31</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb44-40"><a href="#cb44-40" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">32</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"LowQ"</span>,</span>
<span id="cb44-41"><a href="#cb44-41" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">33</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"LowQ"</span>)</span>
<span id="cb44-42"><a href="#cb44-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-43"><a href="#cb44-43" aria-hidden="true" tabindex="-1"></a>    b6_f420_combined<span class="sc">$</span>cell_type_final <span class="ot">&lt;-</span> <span class="fu">Idents</span>(b6_f420_combined)</span>
<span id="cb44-44"><a href="#cb44-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-45"><a href="#cb44-45" aria-hidden="true" tabindex="-1"></a>    b6_f420_assignments <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cell_type_final =</span> b6_f420_combined<span class="sc">$</span>cell_type_final)</span>
<span id="cb44-46"><a href="#cb44-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(b6_f420_assignments) <span class="ot">&lt;-</span> <span class="fu">names</span>(b6_f420_combined<span class="sc">$</span>cell_type_final)</span>
<span id="cb44-47"><a href="#cb44-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-48"><a href="#cb44-48" aria-hidden="true" tabindex="-1"></a>    qs<span class="sc">::</span><span class="fu">qsave</span>(b6_f420_assignments, <span class="at">file =</span> <span class="st">"misc/b6_f420_assignments.qs"</span>)</span>
<span id="cb44-49"><a href="#cb44-49" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="create-publication-plots-and-save" class="level3">
<h3 class="anchored" data-anchor-id="create-publication-plots-and-save">Create publication plots and save</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove tumor cells, low quality cells, and erythrocytes</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(b6_f420_combined) <span class="ot">&lt;-</span> b6_f420_combined<span class="sc">$</span>cell_type_final</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>b6_f420_combined <span class="ot">&lt;-</span> <span class="fu">subset</span>(b6_f420_combined,</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">idents =</span> <span class="fu">c</span>(<span class="st">"LowQ"</span>, <span class="st">"Erythro"</span>, <span class="st">"F420"</span>),</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">invert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot and save</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="fu">r_dim_plot</span>(b6_f420_combined, <span class="st">"F420 Stroma"</span>,</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">"cell_type_final"</span>,</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">split.by =</span> <span class="st">"sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_4-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/combined_b6_f420.pdf"</span>,</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>qs<span class="sc">::</span><span class="fu">qsave</span>(b6_f420_combined, <span class="st">"output/rdata/b6_f420_combined.qs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="create-frequency-plots-of-stromal-and-immune-cells" class="level3">
<h3 class="anchored" data-anchor-id="create-frequency-plots-of-stromal-and-immune-cells">Create frequency plots of stromal and immune cells</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(b6_f420_combined,</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">idents =</span> <span class="fu">c</span>(<span class="st">"Mono"</span>, <span class="st">"Mac"</span>, <span class="st">"DC"</span>, <span class="st">"NK/T"</span>, <span class="st">"B"</span>, <span class="st">"Granulo"</span>, <span class="st">"Alv Mac"</span>))<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(sample, <span class="at">fill =</span> cell_type_final)) <span class="sc">+</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_5-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/b6_immune.pdf"</span>,</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="dv">4</span>,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(b6_f420_combined,</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">idents =</span> <span class="fu">c</span>(</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Endo"</span>,</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Peri"</span>,</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"SMC"</span>,</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Fibroblast"</span>,</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Lower Airway"</span>,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Upper Airway"</span>,</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Meso"</span>,</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Adipo"</span>)</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>    )<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(sample, <span class="at">fill =</span> cell_type_final)) <span class="sc">+</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/b6_f420_5-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/b6_stroma.pdf"</span>,</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="dv">4</span>,</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="balbc-and-k7m2" class="level2">
<h2 class="anchored" data-anchor-id="balbc-and-k7m2">BALBC and K7M2</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>sobj_list <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(<span class="st">"output/rdata/sobj_list.qs"</span>)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge normal and tumor bearing lung for F420 and process into same UMAP space</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>balb_k7m2_combined <span class="ot">&lt;-</span> <span class="fu">merge</span>(sobj_list[[<span class="st">"BALBC"</span>]],</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> sobj_list[[<span class="st">"K7M2"</span>]],</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">add.cell.ids =</span> <span class="fu">c</span>(<span class="st">"BALBC"</span>, <span class="st">"K7M2"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">SCTransform</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">RunPCA</span>(<span class="at">npcs =</span> <span class="dv">30</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">seed.use =</span> <span class="dv">111</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FindNeighbors</span>(<span class="at">k.param =</span> <span class="dv">30</span>, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FindClusters</span>(<span class="at">resolution  =</span> <span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cell attributes from input UMI matrix: log_umi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Variance stabilizing transformation of count matrix of size 19480 by 22729</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Model formula is y ~ log_umi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Get Negative Binomial regression parameters per gene</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using 2000 genes, 5000 cells</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>There are 1 estimated thetas smaller than 1e-07 - will be set to 1e-07</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 106 outliers - those will be ignored in fitting/regularization step</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Second step: Get residuals using fitted parameters for 19480 genes</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |====                                                                  |   5%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  79%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing corrected count matrix for 19480 genes</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |====                                                                  |   5%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  79%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating gene attributes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Wall clock passed: Time difference of 1.843162 mins</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Determine variable features</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Place corrected count matrix in counts slot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering data matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Set default assay to SCT</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  Lyz2, Cxcl2, Il1b, Apoe, Ccl6, C1qa, Arg1, Spp1, Ftl1, Cd74 
       C1qb, Cd14, Wfdc17, Ifi27l2a, Cybb, Ccl9, C1qc, Fth1, Ctss, Ccl4 
       Ctsd, Fcer1g, Tyrobp, H2-Aa, Fn1, H2-Eb1, Pf4, Chil3, Cxcl1, Ctsl 
Negative:  Ramp2, Cldn5, Egfl7, Calcrl, Epas1, Cdh5, Ctla2a, Tmem100, Hpgd, Clec14a 
       Ly6a, Adgrf5, Igfbp7, Gpihbp1, Emp2, Tspan7, Ptprb, Ly6c1, Icam2, Scn7a 
       Kdr, Pecam1, Fmo1, Stmn2, Plvap, Eng, Cyp4b1, Ace, Slc9a3r2, Acvrl1 
PC_ 2 
Positive:  Wfdc2, Sftpb, Sftpa1, Sftpd, Sftpc, Cldn3, Cxcl15, Cbr2, Slc34a2, Chil1 
       Krt8, Cldn18, Napsa, Scd1, Krt18, Ager, Lcn2, Atp1b1, Lamp3, Chchd10 
       S100a4, Muc1, Ces1d, Hc, Ankrd1, Sec14l3, S100a6, Ptprf, S100g, Col1a2 
Negative:  Cxcl2, Apoe, Il1b, Arg1, C1qa, Ccl6, C1qb, Wfdc17, Ctsl, Spp1 
       Pf4, Cd14, C1qc, Ftl1, Ccl9, Ccl4, Fn1, Lyz2, Il1a, Ramp2 
       Cldn5, Ctsd, Ier3, Ctsb, Fcer1g, Ctss, Ccrl2, Ifi27l2a, Srgn, Ccl24 
PC_ 3 
Positive:  Wfdc2, Cbr2, Apoe, Cldn3, Sftpb, Lyz2, Sftpa1, Sftpc, Sftpd, Cxcl15 
       Krt8, Slc34a2, Cxcl2, Arg1, Chil1, C1qa, Krt18, Lcn2, Cldn18, C1qb 
       Napsa, Chchd10, Ager, Sec14l3, Il1b, Pf4, Lamp3, Wfdc17, Ctsl, C1qc 
Negative:  S100a4, Ankrd1, Cd3g, S100a6, Trbc2, Ms4a4b, Trbc1, Crabp2, Il1rl1, Nkg7 
       Cd3d, Col1a2, Rpl39l, Lgals1, Rps24, Ccr7, Igkc, Ccl5, Icos, Satb1 
       Tnfrsf18, Ikzf2, Tnfrsf4, Vps37b, Rpl12, Cd79a, Ptprcap, Rpl13, Rps27, Il2rb 
PC_ 4 
Positive:  S100a6, Apoe, S100a4, Ccl2, C1qa, Lgals1, Col1a2, Arg1, Ankrd1, C1qb 
       Ifitm3, Ifi27l2a, Ccl7, Bgn, C1qc, Cxcl1, Sparc, Pf4, Ctsl, Fn1 
       Crabp2, Il1rl1, Mif, Plac8, Cald1, Ier3, Mgp, Tpm1, Col1a1, Rpl39l 
Negative:  Lpl, Chil3, Plet1, Cybb, Lyz2, Ear2, Fabp1, Abcg1, Ear1, Atp6v0d2 
       Sirpa, Ccl6, Il18, Krt79, Ctsk, Laptm5, Ctsd, Mrc1, Wfdc21, Cd9 
       Slc7a2, Slpi, Gdf15, Ccnd2, Nceh1, Tlr2, Gm42418, Sgk1, Ftl1, Car4 
PC_ 5 
Positive:  Cd74, H2-Aa, H2-Eb1, H2-Ab1, Napsa, Sftpc, Sftpa1, Sftpb, Sftpd, Cxcl15 
       Slc34a2, Chil1, Lcn2, Scd1, Lamp3, Cd52, Ager, Cst3, Cldn18, Hc 
       Muc1, Wfdc2, Igkc, S100g, Lpcat1, Ccr7, Cd79a, Atp1b1, Ctsh, Sfta2 
Negative:  S100a6, Col1a2, Lpl, S100a4, Ankrd1, Ccl6, Chil3, Bgn, Cybb, Crabp2 
       Plet1, Cxcl1, Lgals1, Aldh1a1, Il1rl1, Rarres2, Ftl1, Cald1, Mgp, Ctsd 
       Col1a1, Rpl39l, Sparc, Ear2, Gsn, Mt1, Timp3, Ctgf, Fam183b, Col3a1 </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:48:45 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:48:45 Read 22729 rows and found 30 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:48:45 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:48:45 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
15:48:49 Writing NN index file to temp file /tmp/RtmpwXbYzE/file12ad891b9c7b19
15:48:49 Searching Annoy index using 1 thread, search_k = 3000
15:48:57 Annoy recall = 100%
15:48:58 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
15:49:00 Found 2 connected components, falling back to 'spca' initialization with init_sdev = 1
15:49:00 Using 'irlba' for PCA
15:49:00 PCA: 2 components explained 30.11% variance
15:49:00 Scaling init to sdev = 1
15:49:00 Commencing optimization for 200 epochs, with 961250 positive edges
15:49:10 Optimization finished
Computing nearest neighbor graph
Computing SNN</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 22729
Number of edges: 762053

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9775
Number of communities: 34
Elapsed time: 1 seconds</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check identities from SingleR calls and other meta</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(balb_k7m2_combined, <span class="at">group.by =</span> <span class="st">"seurat_clusters"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(balb_k7m2_combined, <span class="at">group.by =</span> <span class="st">"cell_type"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(balb_k7m2_combined, <span class="at">features =</span> <span class="st">"cell_score"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-3.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(balb_k7m2_combined, <span class="at">features =</span> <span class="st">"nCount_RNA"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-4.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(balb_k7m2_combined, <span class="at">features =</span> <span class="st">"Col1a1"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-5.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">r_feature_plot</span>(balb_k7m2_combined, <span class="at">features =</span> <span class="st">"Col1a2"</span>, <span class="at">split.by =</span> <span class="st">"sample"</span>) <span class="sc">+</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-6.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename clusters based on the calls</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">file.exists</span>(<span class="st">"misc/balb_k7m2_assignments.qs"</span>)) {</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>    balb_k7m2_combined <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(balb_k7m2_combined,</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>        qs<span class="sc">::</span><span class="fu">qread</span>(<span class="st">"misc/balb_k7m2_assignments.qs"</span>))</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># This list is manually curated from the SingleR assignments and scores</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If run again, this needs to be checked for accuracy, as clusters may change</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>    balb_k7m2_combined <span class="ot">&lt;-</span> <span class="fu">RenameIdents</span>(balb_k7m2_combined,</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">0</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">4</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Endo"</span>,</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">5</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"DC"</span>,</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">6</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"B"</span>,</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">7</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"NK/T"</span>,</span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">8</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"K7M2"</span>,</span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">9</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Lower Airway"</span>,</span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">10</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"NK/T"</span>,</span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">11</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"K7M2"</span>,</span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">12</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Granulo"</span>,</span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">13</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mono"</span>,</span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">14</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Endo"</span>,</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">15</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Endo"</span>,</span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">16</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"NK/T"</span>,</span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">17</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"NK/T"</span>,</span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">18</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Granulo"</span>,</span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">19</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"NK/T"</span>,</span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">20</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Mac"</span>,</span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">21</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Fibroblast"</span>,</span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">22</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"SMC"</span>,</span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">23</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Upper Airway"</span>,</span>
<span id="cb88-33"><a href="#cb88-33" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">24</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Lower Airway"</span>,</span>
<span id="cb88-34"><a href="#cb88-34" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">25</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb88-35"><a href="#cb88-35" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">26</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Endo"</span>,</span>
<span id="cb88-36"><a href="#cb88-36" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">27</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Erythro"</span>,</span>
<span id="cb88-37"><a href="#cb88-37" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">28</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Adipo"</span>,</span>
<span id="cb88-38"><a href="#cb88-38" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">29</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Alv Mac"</span>,</span>
<span id="cb88-39"><a href="#cb88-39" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">30</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"LowQ"</span>,</span>
<span id="cb88-40"><a href="#cb88-40" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">31</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"DC"</span>,</span>
<span id="cb88-41"><a href="#cb88-41" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">32</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"LowQ"</span>,</span>
<span id="cb88-42"><a href="#cb88-42" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">33</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"LowQ"</span>,</span>
<span id="cb88-43"><a href="#cb88-43" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">34</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Peri"</span>)</span>
<span id="cb88-44"><a href="#cb88-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-45"><a href="#cb88-45" aria-hidden="true" tabindex="-1"></a>    balb_k7m2_combined<span class="sc">$</span>cell_type_final <span class="ot">&lt;-</span> <span class="fu">Idents</span>(balb_k7m2_combined)</span>
<span id="cb88-46"><a href="#cb88-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-47"><a href="#cb88-47" aria-hidden="true" tabindex="-1"></a>    balb_k7m2_assignments <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cell_type_final =</span> balb_k7m2_combined<span class="sc">$</span>cell_type_final)</span>
<span id="cb88-48"><a href="#cb88-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(balb_k7m2_assignments) <span class="ot">&lt;-</span> <span class="fu">names</span>(balb_k7m2_combined<span class="sc">$</span>cell_type_final)</span>
<span id="cb88-49"><a href="#cb88-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-50"><a href="#cb88-50" aria-hidden="true" tabindex="-1"></a>    qs<span class="sc">::</span><span class="fu">qsave</span>(balb_k7m2_assignments, <span class="at">file =</span> <span class="st">"misc/balb_k7m2_assignments.qs"</span>)</span>
<span id="cb88-51"><a href="#cb88-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb88-52"><a href="#cb88-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-53"><a href="#cb88-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove tumor cells, low quality cells, and erythrocytes</span></span>
<span id="cb88-54"><a href="#cb88-54" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(balb_k7m2_combined) <span class="ot">&lt;-</span> balb_k7m2_combined<span class="sc">$</span>cell_type_final</span>
<span id="cb88-55"><a href="#cb88-55" aria-hidden="true" tabindex="-1"></a>balb_k7m2_combined <span class="ot">&lt;-</span> <span class="fu">subset</span>(balb_k7m2_combined,</span>
<span id="cb88-56"><a href="#cb88-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">idents =</span> <span class="fu">c</span>(<span class="st">"LowQ"</span>, <span class="st">"Erythro"</span>, <span class="st">"K7M2"</span>),</span>
<span id="cb88-57"><a href="#cb88-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">invert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb88-58"><a href="#cb88-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-59"><a href="#cb88-59" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot and save</span></span>
<span id="cb88-60"><a href="#cb88-60" aria-hidden="true" tabindex="-1"></a><span class="fu">r_dim_plot</span>(balb_k7m2_combined, <span class="st">"K7M2 Stroma"</span>,</span>
<span id="cb88-61"><a href="#cb88-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">"cell_type_final"</span>,</span>
<span id="cb88-62"><a href="#cb88-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">split.by =</span> <span class="st">"sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-7.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/combined_balb_k7m2.pdf"</span>,</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>qs<span class="sc">::</span><span class="fu">qsave</span>(balb_k7m2_combined, <span class="st">"output/rdata/balb_k7m2_combined.qs"</span>)</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create frequency plots of stromal and immune cells</span></span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(balb_k7m2_combined,</span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">idents =</span> <span class="fu">c</span>(<span class="st">"Mono"</span>, <span class="st">"Mac"</span>, <span class="st">"DC"</span>, <span class="st">"NK/T"</span>, <span class="st">"B"</span>, <span class="st">"Granulo"</span>, <span class="st">"Alv Mac"</span>))<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(sample, <span class="at">fill =</span> cell_type_final)) <span class="sc">+</span></span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-8.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/balb_immune.pdf"</span>,</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="dv">4</span>,</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(balb_k7m2_combined,</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">idents =</span> <span class="fu">c</span>(<span class="st">"Endo"</span>, <span class="st">"Peri"</span>, <span class="st">"SMC"</span>, <span class="st">"Fibroblast"</span>, <span class="st">"Lower Airway"</span>, <span class="st">"Upper Airway"</span>))<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(sample, <span class="at">fill =</span> cell_type_final)) <span class="sc">+</span></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="Load-Cluster_files/figure-html/balbc_k7m2-9.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"output/figures/balb_stroma.pdf"</span>,</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="dv">4</span>,</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>