--
title: "Data Analysis for Fibrosis paper"
author: "James Reineke and Matt Cannon"
date: "`r format(Sys.time(), '%m/%d/%Y')`"
output:
    html_document:
        toc: true
        toc_float: true
        toc_depth: 5
        number_sections: false
        code_folding: hide
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    tidy = TRUE,
    echo = TRUE,
    cache = TRUE,
    collapse = TRUE,
    tidy.opts = list(width.cutoff = 95),
    cache.lazy = FALSE
)
```

# Sample summary

- Our data
    - stuff here
- GSE141259
    - Longitudinal single-cell transcriptomics analysis of mouse lung upon bleomycin-induced injury
    - PMID: 32678092
    - DropSeq
    - Has three standard gz files plus metadata
- GSE145031
    - Inflammatory Signals induce AT2 Cell-Derived Damage-Associated Transient Progenitors that Mediate Alveolar Regeneration
    - PMID: 32750316
    - 10X
    - Has three standard gz files
- GSE157995
    - Single-cell RNA-sequencing of bleomycin-injured young and old mice
    - PMID: 35389887
    - 10X
    - Has csv.gz files with count matrices
    - Six samples have .h5 files instead
    - No metadata included and the code is hand annotated
- GSE201698
    - Time-course cellular interactome analysis by single-cell RNA-seq in bleomycin-induced pulmonary fibrosis
    - not published
    - BD Rhapsody platform
    - Has txt.gz files with count matrices
- GSE151374
    - The Anti-Fibrotic Drug Nintedanib Promotes Expansion of Lung Macrophages with a Distinct Transcriptional Repair Program in Bleomycin-Exposed Mice
    - PMID: 36227799
    - Think it's 10x - Not sure - paper not available
    - Has .h5 files

# Load packages
```{r lib}
library(cowplot)
library(cluster)
library(tidyverse)
library(ggrepel)
library(patchwork)
library(reticulate)
library(Seurat)
library(sctransform)
library(rrrSingleCellUtils)
library(msigdbr)
# Set random generator seed to facilitate reproducibility
set.seed(888)
```

## Make up directory structure
```{bash mkdirs, eval=TRUE}
for directoryName in \
  output \
  output/figures \
  output/rdata \
  output/de
do
  if [ ! -d ${directoryName} ]
  then
    mkdir -p ${directoryName}
  fi
done
```


### Get cell type reference datasets
GSE151974
```{r mouse_refs}
ref_path <- "/home/gdrobertslab/lab/GenRef/sc_ref_datasets/mouse"
mouse_lung_ref <- qs::qread(paste0(ref_path, "/GSE151974/mouse_lung_ref.qs"))
mouse_immune <- celldex::ImmGenData()
# This label is not informative
mouse_immune <-
    mouse_immune[mouse_immune$label.main != "Stromal cells", ]
mouse_rna <- celldex::MouseRNAseqData()
```

## Function to use for cell annotation
```{r annotate_cells_fun}
annot_cells <- function(sobject) {
    cell_assign <-
        SingleR::SingleR(as.SingleCellExperiment(sobject),
                         ref = list(GetAssayData(mouse_lung_ref),
                                    mouse_immune,
                                    mouse_rna),
                         labels = list(mouse_lung_ref$cell_type,
                                       mouse_immune$label.main,
                                       mouse_rna$label.fine),
                         aggr.ref = TRUE)
    sobject$cell_type <- cell_assign$labels
    sobject$cell_score <- cell_assign$scores %>%
        apply(MARGIN = 1, function(x) max(x, na.rm = TRUE))
    return(sobject)
}
crazy_cols <- function(x, seed = 1337) {
    set.seed(seed)
    sample(rainbow(n = length(unique(x))))
}
calc_logfc <- function(sobj,
                       group_var,
                       group_1,
                       group_2,
                       epsilon = 1,
                       assay = "SCT") {
    all_obs_exp <-
        AverageExpression(sobj,
                          group.by = group_var,
                          assays = assay)[[1]] %>%
        as.data.frame()

    log_fc <-
        tibble(log_fc = log2((all_obs_exp[[group_1]] + epsilon) /
                             (all_obs_exp[[group_2]] + epsilon)),
               gene = rownames(all_obs_exp))

    return(log_fc)
}
```


# Load raw data
```{r load_raw, cache.vars = '', eval=TRUE}
sample_list <- read_tsv("misc/sample_cutoffs.txt", show_col_types = FALSE)

end_path <- "/filtered_feature_bc_matrix"
if (grepl("r1pl", Sys.info()[["nodename"]])) {
    data_path <- "/home/gdrobertslab/lab/Counts/"
    path_use <- "cluster_path"
} else {
    data_path <- "/Applications/scRNA_seq raw files/"
    path_use <- "James_path"
}

#for (i in seq_len(nrow(sample_list))) {
sobj_list <-
    parallel::mclapply(seq_len(nrow(sample_list)),
                       mc.cores = 4,
                       function(i) {
    obj_name <- sample_list$obj_name[i]
    message(obj_name)

    sobj <-
        tenx_load_qc(paste0(data_path,
                            sample_list[[path_use]][i],
                            end_path),
                     min_cells = 3,
                     min_features = 200,
                     violin_plot = FALSE)

    # Add metadata to dataset
    for (colname in colnames(sample_list)) {
        sobj[[colname]] <- sample_list[[colname]][i]
    }

    # Add sample name to dataset
    sobj$sample <- obj_name

    cutoff_table <-
        tribble(~"feature",   ~"min_val",                    ~"max_val",
                "nCount_RNA", sample_list$ncount_rna_min[i], sample_list$ncount_rna_max[i],
                "percent.mt", 0,                             sample_list$mt_max[i])

    plotted <-
        feature_hist(sobj,
                     features = c("nCount_RNA", "percent.mt"),
                     cutoff_table = cutoff_table)

    ggsave(paste0("output/figures/feature_hist_",
                  obj_name,
                  ".png"),
           width = 10,
           height = 10,
           plot = plotted)

    sobj <-
        sobj %>%
        subset(nCount_RNA   >= sample_list$ncount_rna_min[i] &
               nCount_RNA   <= sample_list$ncount_rna_max[i] &
               percent.mt   <= sample_list$mt_max[i]) %>%
        process_seurat()

    sobj <- annot_cells(sobj)
    return(sobj)
})

names(sobj_list) <- sample_list$obj_name

qs::qsave(sobj_list,
          file = "output/rdata/sobj_list.qs")
```

# Make Seurat object of all the data
```{r merge_all, dependson='load_raw', eval=TRUE}
sobj_list <- qs::qread("output/rdata/sobj_list.qs")
all_sobj <-
    merge(sobj_list[[1]],
          sobj_list[2:length(sobj_list)]) %>%
    SCTransform(vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    RunPCA(verbose = FALSE, assay = "SCT") %>%
    FindNeighbors(verbose = FALSE) %>%
    FindClusters(verbose = FALSE) %>%
    RunUMAP(verbose = FALSE, dims = 1:30)

all_dim <-
    DimPlot(all_sobj,
            group.by = c("obj_name", "cell_type"),
            label = TRUE,
            repel = TRUE,
            cols = c(plot_cols, crazy_cols(all_sobj$cell_type, seed = 1341))) +
    NoLegend()

ggsave("output/figures/all_dim.pdf",
       width = 25,
       height = 8)

# Grab just immune cells

mac_types <-
    c("Int Mf",
      "Macrophages",
      "Macrophages activated")

Idents(all_sobj) <- "cell_type"

sub_sobj <-
    subset(all_sobj,
           idents = mac_types)

source("scripts/recurl.R")

immune_recurl <-
    recurluster(sub_sobj,
                parallel = TRUE,
                do_plots = FALSE)

DefaultAssay(immune_recurl) <- "SCT"
immune_recurl <-
    immune_recurl %>%
    SCTransform(vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    RunPCA(verbose = FALSE, assay = "SCT") %>%
    FindNeighbors(verbose = FALSE) %>%
    FindClusters(verbose = FALSE) %>%
    RunUMAP(verbose = FALSE, dims = 1:30)

qs::qsave(immune_recurl,
          file = "output/rdata/immune_recurl.qs")

macs_dim <-
    DimPlot(immune_recurl,
            group.by = c("obj_name", "clust_1", "clust_2"),
            label = TRUE,
            repel = TRUE,
            cols = c(plot_cols, crazy_cols(immune_recurl$clust_2, seed = 1341)))

ggsave("output/figures/mac_dim2.pdf",
       width = 25,
       height = 8)

cluster_counts <-
    table(immune_recurl$obj_name, immune_recurl$clust_2) %>%
    as.data.frame() %>%
    dplyr::rename("Sample" = "Var1",
                  "Cluster" = "Var2",
                  "Count" = "Freq") %>%
    group_by(Cluster) %>%
    arrange(desc(Count), .by_group = TRUE) %>%
    ggplot(aes(y = Cluster, x = Sample, fill = log10(Count))) +
    geom_tile() +
    geom_text(aes(label = Count), color = "white")

ggsave("output/figures/mac_cluster_counts2.pdf",
       width = 8,
       height = 8)

Idents(immune_recurl) <- "clust_2"
```

## Look at inflammatory genes in macs
```{r macs_inflam, dependson='merge_all', eval=TRUE}
inflam_genes <-
    c("H2-Eb1", "H2-Ab1", "Cd86", "Il1a", "Il1b", "Cxcl1", "Cxcl2")

anti_inflam_genes <-
    c("Gpnmb", "Fabp5", "Anpep", "C1qa", "C1qb", "Fcrls", "Trem2", "Fn1", "Spp1")

infl_macs <-
    FeaturePlot(immune_recurl,
                features = inflam_genes)

ggsave("output/figures/mac_inflam.pdf",
       width = 8,
       height = 8,
       plot = infl_macs)

infl_macs_vln <-
    VlnPlot(immune_recurl,
            features = inflam_genes,
            group.by = "clust_2",
            cols = c(plot_cols, crazy_cols(immune_recurl$clust_2)),
            pt.size = 0,
            adjust = 0.5)

ggsave("output/figures/mac_inflam_vln.pdf",
       width = 15,
       height = 8,
       plot = infl_macs_vln)

anti_imm_macs <-
    FeaturePlot(immune_recurl,
                features = anti_inflam_genes)

ggsave("output/figures/mac_anti_inflam.pdf",
       width = 8,
       height = 8,
       plot = anti_imm_macs)

anti_imm_macs_vln <-
    VlnPlot(immune_recurl,
            features = anti_inflam_genes,
            group.by = "clust_2",
            cols = c(plot_cols, crazy_cols(immune_recurl$clust_2)),
            pt.size = 0,
            adjust = 0.5)

ggsave("output/figures/mac_anti_inflam_vln.pdf",
       width = 15,
       height = 8,
       plot = anti_imm_macs_vln)

immune_recurl <-
    AddModuleScore(immune_recurl,
                   features = list(inflam_genes),
                   name = "inflam_module")

immune_recurl <-
    AddModuleScore(immune_recurl,
                   features = list(anti_inflam_genes),
                   name = "anti_inflam_module")

qs::qsave(immune_recurl,
          file = "output/rdata/immune_recurl.qs")

imm_mod_vln <-
    VlnPlot(immune_recurl,
            features = c("inflam_module1", "anti_inflam_module1"),
            group.by = "clust_2",
            ncol = 1,
            cols = c(plot_cols, crazy_cols(immune_recurl$clust_2)),
            pt.size = 0,
            adjust = 0.5)

ggsave("output/figures/mac_inflam_mod_vln.pdf",
       width = 12,
       height = 8,
       plot = imm_mod_vln)
```


# run GSEA on all macs between primary and met
```{r }
tumor_only <-
    subset(immune_recurl,
           obj_name %in% c("K7M2", "F420", "tibia_F420", "tibia_K7M2"))

tumor_only$comp <-
    if_else(tumor_only$obj_name == "tibia_F420" |
            tumor_only$obj_name == "tibia_K7M2",
            "Primary",
            "Met")

qs::qsave(tumor_only,
          file = "output/rdata/macs_tumor_only.qs")

prim_met_logfc <-
    calc_logfc(tumor_only,
               group_var = "comp",
               group_1 = "Primary",
               group_2 = "Met") %>%
    arrange(desc(log_fc)) %>%
    pull(log_fc, name = gene)

cat_tib <-
    tribble(~category,          ~subcat,    ~label,
            "C2",               "CP:KEGG",  "KEGG",
            "C3",               "TFT:GTRD", "Transcription factors",
            "C5",               "GO:BP",    "GO Biological Process",
            "C5",               "GO:MF",    "GO Molecular Function",
            "C8",               "",         "Cell type")

gsea_out <-
    parallel::mclapply(seq_len(nrow(cat_tib)),
                       mc.cores = 5,
                       function(i) {
        message(cat_tib$label[i])
        kegg_ref <-
            msigdbr::msigdbr(species = "Mus musculus",
                            category = cat_tib$category[i],
                            subcategory = cat_tib$subcat[i]) %>%
                split(x = .$gene_symbol, f = .$gs_name)

        fgsea::fgseaMultilevel(kegg_ref,
                               prim_met_logfc,
                               minSize = 15,
                               maxSize = 500,
                               nPerm = 1000) %>%
            arrange(desc(NES)) %>%
            mutate(pathway = str_replace_all(pathway, "_", " "),
                   category = cat_tib$category[i],
                   sub_cat = cat_tib$subcat[i],
                   cat_label = cat_tib$label[i]) %>%
            filter(padj < 0.05)
    }) %>%
    bind_rows()

top_paths <-
    gsea_out %>%
    group_by(cat_label, NES > 0) %>%
    arrange(desc(abs(NES)), .by_group = TRUE) %>%
    slice_head(n = 10) %>%
    pull(pathway) %>%
    str_wrap(width = 80)

gsea_dots <-
    gsea_out %>%
    filter(pathway %in% top_paths) %>%
    mutate(pathway = fct_reorder(pathway, NES)) %>%
    ggplot(aes(y = pathway,
               x = NES,
               size = -log10(padj))) +
    geom_point() +
    theme(axis.text.y = element_text(size = 4))

ggsave("output/figures/gsea_dots_macs_prim_met.pdf",
       width = 8,
       height = 15,
       plot = gsea_dots)
```

## GSEA comparing met to primary for each cluster
```{r kegg_gsea, dependson='merge_all', eval=TRUE}
min_cell_comp <- 10
all_logfc <-
    sapply(unique(tumor_only$clust_2),
           USE.NAMES = TRUE,
           function(x) {
        test_data <-
            subset(tumor_only,
                   clust_2 == x)

        if (min(table(test_data$comp)) > min_cell_comp &&
            length(unique(test_data$comp)) == 2) {
            output <-
                calc_logfc(test_data,
                            group_var = "comp",
                            group_1 = "Primary",
                            group_2 = "Met") %>%
                arrange(desc(log_fc)) %>%
                pull(log_fc, name = gene)
        } else {
            output <- NULL
        }
        return(output)
    })

all_logfc <- all_logfc[!all_logfc %in% list(NULL)]

# GSEA
set.seed(1337)

cat_tib <-
    tribble(~category,          ~subcat,    ~label,
            "C2",               "CP:KEGG",  "KEGG",
            "C3",               "TFT:GTRD", "Transcription factors",
            "C5",               "GO:BP",    "GO Biological Process",
            "C5",               "GO:MF",    "GO Molecular Function",
            "C8",               "",         "Cell type")

gsea_out <-
    parallel::mclapply(seq_len(nrow(cat_tib)),
                       mc.cores = 5,
                       function(i) {
        message(cat_tib$label[i])
        kegg_ref <-
            msigdbr::msigdbr(species = "Mus musculus",
                            category = cat_tib$category[i],
                            subcategory = cat_tib$subcat[i]) %>%
                split(x = .$gene_symbol, f = .$gs_name)

            parallel::mclapply(names(all_logfc),
                            mc.cores = 5,
                            function(x) {
                fgsea::fgseaMultilevel(kegg_ref,
                                       all_logfc[[x]],
                                       minSize = 15,
                                       maxSize = 500,
                                       nPerm = 1000) %>%
                    arrange(desc(NES)) %>%
                    mutate(pathway = str_replace_all(pathway, "_", " "),
                           cluster = x,
                           category = cat_tib$category[i],
                           sub_cat = cat_tib$subcat[i],
                           cat_label = cat_tib$label[i]) %>%
                    filter(padj < 0.05)
            }) %>%
            bind_rows()
    }) %>%
    bind_rows()

lapply(unique(gsea_out$cat_label),
       function(x) {
    top_paths <-
        gsea_out %>%
        filter(cat_label == x) %>%
        group_by(cluster) %>%
        arrange(desc(abs(NES)), .by_group = TRUE) %>%
        slice_head(n = 20) %>%
        pull(pathway) %>%
        str_wrap(width = 80)
    gsea_out %>%
        filter(pathway %in% top_paths) %>%
        pivot_wider(names_from = pathway,
                    values_from = NES,
                    id_cols = c(cluster),
                    values_fill = 0) %>%
        column_to_rownames("cluster") %>%
        t() %>%
        pheatmap::pheatmap(main = x,
                           filename = paste0("output/figures/gsea_",
                                             x,
                                             ".pdf"),
                           width = 8,
                           height = 12,
                           fontsize_row = 6)
    })
```

# Confirm directionality of genes
```{r kegg_ribo, dependson='load_raw', eval=TRUE}
kegg_ribo_genes <-
    msigdbr::msigdbr(species = "Mus musculus",
                     category = "C2",
                     subcategory = "KEGG") %>%
    filter(gs_name == "KEGG_RIBOSOME") %>%
    split(x = .$gene_symbol, f = .$gs_name)

tumor_only <-
    AddModuleScore(tumor_only,
                   features = kegg_ribo_genes,
                   name = "kegg_ribo_module")

tumor_only %>%
    subset(clust_2 %in% c("clust_0.0",
                          "clust_1.3",
                          "clust_1.2",
                          "clust_3.1",
                          "clust_3.0")) %>%
    VlnPlot(split.by = "sample",
            features = "kegg_ribo_module1",
            group.by = "clust_2")

plot_genes <-
    gsea_out %>%
    group_by(cluster) %>%
    filter(p_val_adj <= 0.05) %>%
    arrange(avg_log2FC, .by_group = TRUE) %>%
    slice_head(n = 2) %>%
    pull(gene) %>%
    unique()

FeaturePlot(immune_recurl,
            features = plot_genes)

DotPlot(immune_recurl, group.by = c("obj_name", "clust_1", "cell_type"),
        label = TRUE) +
    NoLegend()
```

## Run GSEA to compare each cluster to all others
```{r kegg_gsea2, dependson='load_raw', eval=TRUE}
all_logfc_clust <-
    sapply(unique(tumor_only$clust_2),
           USE.NAMES = TRUE,
           function(x) {
        test_data <- tumor_only
        test_data$comp <-
            if_else(test_data$clust_2 == x,
                    x,
                    "other")

        output <-
            calc_logfc(test_data,
                        group_var = "comp",
                        group_1 = x,
                        group_2 = "other") %>%
            arrange(desc(log_fc)) %>%
            pull(log_fc, name = gene)
        return(output)
    })

set.seed(1337)
gsea_out_clust <-
    parallel::mclapply(seq_len(nrow(cat_tib)),
                       mc.cores = 5,
                       function(i) {
        message(cat_tib$label[i])
        kegg_ref <-
            msigdbr::msigdbr(species = "Mus musculus",
                            category = cat_tib$category[i],
                            subcategory = cat_tib$subcat[i]) %>%
                split(x = .$gene_symbol, f = .$gs_name)

            parallel::mclapply(names(all_logfc),
                            mc.cores = 5,
                            function(x) {
                fgsea::fgseaMultilevel(kegg_ref,
                                       all_logfc[[x]],
                                       minSize = 15,
                                       maxSize = 500,
                                       nPerm = 1000) %>%
                    arrange(desc(NES)) %>%
                    mutate(pathway = str_replace_all(pathway, "_", " "),
                           cluster = x,
                           category = cat_tib$category[i],
                           sub_cat = cat_tib$subcat[i],
                           cat_label = cat_tib$label[i]) %>%
                    filter(padj < 0.05)
            }) %>%
            bind_rows()
    }) %>%
    bind_rows()

lapply(unique(gsea_out_clust$cat_label),
       function(x) {
    top_paths <-
        gsea_out_clust %>%
        filter(cat_label == x) %>%
        group_by(cluster) %>%
        arrange(desc(abs(NES)), .by_group = TRUE) %>%
        slice_head(n = 20) %>%
        pull(pathway) %>%
        str_wrap(width = 80)
    gsea_out_clust %>%
        filter(pathway %in% top_paths) %>%
        pivot_wider(names_from = pathway,
                    values_from = NES,
                    id_cols = c(cluster),
                    values_fill = 0) %>%
        column_to_rownames("cluster") %>%
        t() %>%
        pheatmap::pheatmap(main = x,
                           filename = paste0("output/figures/gsea_by_cluster_",
                                             x,
                                             ".pdf"),
                           width = 8,
                           height = 12,
                           fontsize_row = 6)
    })
```

# Make up Seurat objects with subsets of the data

## B6 and F420
```{r b6_f420, dependson='load_raw', eval=TRUE}
sobj_list <- qs::qread("output/rdata/sobj_list.qs")

###scTransform merged data sets to include all genes in scaled data
b6_f420_combined <-
    merge(sobj_list[["C57BL6"]],
          y = sobj_list[["F420"]],
          add.cell.ids = c("C57BL6", "F420")) %>%
    SCTransform(vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(resolution = 0.2,
                   run_umap_dims = 1:30)

b6_f420_combined <-
    RenameIdents(b6_f420_combined,
                 `0` = "Macrophage",
                 `1` = "Alveolar macrophage",
                 `2` = "Endothelial cell",
                 `4` = "T cell",
                 `5` = "Dendritic cell",
                 `6` = "Proximal airway cell",
                 `7` = "Distal airway cell",
                 `8` = "Fibroblast",
                 `9` = "B cell",
                 `10` = "Endothelial cell",
                 `11` = "Dendritic cell",
                 `12` = "Granulocyte",
                 `14` = "F420",
                 `15` = "Erythrocyte",
                 `16` = "Smooth muscle cell",
                 `17` = "Adipocyte",
                 `18` = "Mesothelial cell")

###add cluster identity to metadata
b6_f420_combined$celltype_sample <-
    paste(Idents(b6_f420_combined),
          b6_f420_combined$sample,
          sep = "_")

b6_f420_combined$celltype <- Idents(b6_f420_combined)

DimPlot(b6_f420_combined,
        label = TRUE,
        split.by = "sample",
        pt.size = 1,
        cols = scales::alpha(plot_cols, 0.6)) +
    coord_fixed() +
    theme(axis.text = element_text(size = 5),
          axis.title = element_text(size = 8),
          plot.title = element_text(size = 10,
                                    face = "bold",
                                    hjust = 0.5),
          strip.text = element_text(size = 8, face = "bold"),
          plot.subtitle = element_text(size = 8, hjust = 0.5),
          legend.text = element_text(size = 5),
          legend.title = element_text(size = 6, face = "bold"),
          strip.background = element_blank())

ggsave("output/figures/combined_B6_F420.pdf",
       width = 4,
       height = 4)
```

## BALBC and K7M2
```{r balbc_k7m2, dependson='load_raw', eval=TRUE}
###merge control and met datasets
balbc_k7m2_combined <-
    merge(sobj_list[["BALBC"]],
          y = sobj_list[["K7M2"]],
          add.cell.ids = c("BALBC", "K7M2")) %>%
    SCTransform(vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(resolution = 0.2,
                   run_umap_dims = 1:30)

###Remove dead cells cluster 9  noted by high expression of mitochondrial as well as small clusters 18-20
dead_cell_cluster <- "9"
small_clusters <- c("18", "19", "20")

balbc_k7m2_combined <-
    subset(balbc_k7m2_combined,
           idents = c(dead_cell_cluster,
                      small_clusters),
           invert = T)

balbc_k7m2_combined <-
    RenameIdents(balbc_k7m2_combined,
                 `0` = "Alveolar macrophage",
                 `1` = "Macrophage",
                 `2` = "T cell",
                 `3` = "Endothelial cell",
                 `4` = "K7M2",
                 `5` = "Granulocyte",
                 `6` = "Dendritic cell",
                 `7` = "B cell",
                 `8` = "Distal airway cell",
                 `10` = "Fibroblast",
                 `11` = "Endothelial cell",
                 `12` = "Macrophage",
                 `13` = "T cell",
                 `14` = "Proximal airway cell",
                 `15` = "Alveolar macrophage",
                 `16` = "Endothelial cell",
                 `17` = "Adipocyte")

###add cluster identity to metadata
balbc_k7m2_combined$celltype_sample <-
    paste(Idents(balbc_k7m2_combined),
          balbc_k7m2_combined$sample,
          sep = "_")

balbc_k7m2_combined$celltype <- Idents(balbc_k7m2_combined)
```

# Process the data further

## B6 F420 datasets

### Run DE on B6 F420 combined data sets
This isn't used further. Is it needed or should we save it out to a text file?
```{r de_b6_F420, dependson='b6_f420', eval=TRUE}
initial_markers <-
    FindAllMarkers(b6_f420_combined,
                   only.pos = TRUE,
                   min.pct = 0.25,
                   logfc.threshold = 0.5)
```

### Remove cycling and dead cells
clusters 3 and 13 noted by high expression of mitochondrial and cell cycling genes respectively

We should come up with automated ways to define these - MVC
```{r remove_cycling, dependson='de_b6_F420', eval=TRUE}
b6_f420_combined <-
    subset(b6_f420_combined,
           idents = c("3", "13"),
           invert = TRUE)
```

### Addmodules GSEA with msigdbr

cutoff: logfc > 1 and less than in 15% of other cells, >50% of cells in cluster

Doing this on b6_f420_combined and balbc_k7m2_combined
```{r add_modules, dependson='cluster_assignments', eval=TRUE}
# Not used?
###markers for plotting
# b6_F420_combined.plotmarkers <-
#     c("Itgam", "Fcgr1", "Cd68", "Adgre1", "Ccr2", "Cx3cr1", "Ly6c2", "Itgax",
#       "Mertk", "Marco", "Siglecf", "Egr2", "Bhlhe41", "Fabp1", "Olr1", "Alox5",
#       "Vwf", "Tek", "Acer2", "Cavin3", "Pecam1", "Emcn", "Lyve1", "Ptprb",
#       "Plvap", "Cd3d", "Cd4", "Cd8b1", "Tcf7", "Il7r", "Lef1", "Dusp10", "Itk",
#       "Cd207", "Cd209a", "Ccr7", "Ccl17", "Ccl22", "Wdfy4", "Clec4b1",
#       "H2-DMb1", "Etv3", "Itgae", "Xcr1", "Epcam", "Scgb1a1", "Tuba1a", "Foxj1",
#       "Sftpd", "Lamp3", "Etv5", "Ager", "Aqp5", "Col1a2", "Col3a1", "Col1a1",
#       "Loxl1", "Pdgfra", "Pdgfrb", "Fn1", "Ltbp4", "Bgn", "Cd19", "Cd79a",
#       "Cd79b", "Iglc2", "Ighm", "Pax5", "Cd33", "Csf3r", "S100a9", "Cxcr2",
#       "Mxd1", "Ptafr", "Steap1", "Crabp2", "Wisp1", "Dclk1", "Twist1", "Il1rl1",
#       "Ctgf", "Hbb-bs", "Hba-a1", "Myh11", "Cox4i2", "Pde5a", "Cfd", "Car3",
#       "Adipoq", "Msln")

all_gene_sets <- msigdbr(species = "Mus musculus")

pathways <-
    list("HALLMARK_INFLAMMATORY_RESPONSE" = "Inflammation_features",
         "WP_LUNG_FIBROSIS" = "Lung_fibrosis_features",
         "HALLMARK_TNFA_SIGNALING_VIA_NFKB" = "TNFA_features",
         "REACTOME_DEGRADATION_OF_THE_EXTRACELLULAR_MATRIX" = "ECM_degrad_features",
         "REACTOME_EXTRACELLULAR_MATRIX_ORGANIZATION" = "ECM_org_features",
         "GOBP_RESPONSE_TO_WOUNDING" = "Wound_response_features",
         "HALLMARK_P53_PATHWAY" = "p53_features",
         "FRIDMAN_SENESCENCE_UP" = "senescence_features",
         "GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP" = "m1_features",
         "GOBP_POSITIVE_REGULATION_OF_PHAGOCYTOSIS" = "phagocytosis_features")

for (sample_name in c("b6_f420_combined", "balbc_k7m2_combined")) {
    # temporarily copy the data from the target object so we can loop through
    sobj <- get(sample_name)
    for (pathway in names(pathways)) {
        gene_list <-
            list(filter(all_gene_sets,
                        gs_name == pathway)$gene_symbol)
        sobj <-
            AddModuleScore(sobj,
                           features = gene_list,
                           name = pathways[[pathway]])
    }
    # Put the new data back into the original variable
    assign(sample_name, sobj)
    # Don't need this anymore
    rm(sobj)
}
```

## Subset single cell types for specific analysis

### Subset epithelial cells from B6/F420 cells and re-normalize
```{r subset_epithelial, dependson='add_modules', eval=TRUE}
b6_f420_combined_aec_1 <-
    b6_f420_combined %>%
    subset(idents = "Distal airway cell") %>%
    SCTransform(vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(run_umap_dims = 1:30, resolution = 0.6)

b6_f420_combined_aec_1 <-
    b6_f420_combined_aec_1 %>%
    ### Remove contaminating non-epithelial cells-immune cells
    subset(idents = c("1", "2", "3", "6"), invert = TRUE) %>% # Should this be inverted? wasn't sure - mvc
    ###remove contaminating proximal airway cells
    subset(idents = "5", invert = TRUE) %>%
    ###re-cluster with pure epithelial cell populations
    SCTransform(vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(run_umap_dims = 1:30, resolution = 0.6) %>%
    # Cluster identification based on published data sets-PMIDs
    # 32750316,32661339,32678092
    RenameIdents(`0` = "pAEC2",
                 `1` = "DATP",
                 `2` = "AEC2",
                 `3` = "AEC1",
                 `4` = "pAEC2",
                 `6` = "AEC1",
                 `7` = "cAEC2")
```

#### DE on B6/F420 epithelial cells
This isn't used further, should we save it out to a file?
```{r DE_epithelial, dependson='subset_epithelial', eval=TRUE}
initial_epimarkers <-
    FindAllMarkers(b6_f420_combined_aec_1,
                   only.pos = TRUE,
                   min.pct = 0.25,
                   logfc.threshold = 0.5)


# This isn't used anywhere
# If we want to use this sort of info somewhere we should centralize it to a
# single chunk somewhere up top
# Distalairway_markers <-
#     c("Lcn2", "Lrg1", "Cxcl17", "Glrx", "Krt8", "Sfn", "Krt19", "Cldn4",
#       "Trp53", "Ctgf", "Sftpa1", "Sftpb", "Sftpc", "Lpcat1", "Etv5", "Abca3",
#       "Pdpn", "Aqp5", "Hopx", "Cav1", "Ager", "Akap5", "Clic5", "Top2a",
#       "Mki67", "Cdk1")
```

#### F420 macrophages
```{r f420_macs, dependson='subset_epithelial', eval=TRUE}
f420_macs <-
    b6_f420_combined_aec_1 %>%
    subset(idents = "PUT STUFF HERE TO GET MACS") %>%
    SCTransform(f420_macs,
                vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(run_umap_dims = 1:30, resolution = 0.4)

# Not used anywhere
# mac.markers <-
#     c("Itgam","Itgax", "Cd14", "Fcgr3", "Fcgr1", "Cd68", "Adgre1", "Ccr2",
#       "Cx3cr1", "Ly6c1", "Ly6c2", "H2-Ab1", "H2-DMb1")
```

### Epithelial cells from BALB/c K7M2
```{r subset_epithelial_balbc, dependson='add_modules', eval=TRUE}
balbc_k7m2_combined_aec <-
    subset(balbc_k7m2_combined, idents = "Distal airway cell") %>%
    SCTransform(vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(run_umap_dims = 1:30, resolution = 0.6)

balbc_k7m2_combined_aec <-
    balbc_k7m2_combined_aec %>%
    ###remove contaminating non-epithelial cells-immune cells
    subset(idents = c("2", "3", "4", "5", "6", "7")) %>%
    SCTransform(balbc_k7m2_combined_aec,
                vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(run_umap_dims = 1:30, resolution = 0.6) %>%
    # Cluster identification based on published data sets-PMIDs
    # 32750316,32661339,32678092
    RenameIdents(`0` = "pAEC2",
                 `1` = "DATP",
                 `2` = "AEC2",
                 `3` = "AEC2",
                 `4` = "AEC1",
                 `5` = "cAEC2")

balbc_k7m2_combined.AEC_1$celltype_sample <-
    paste(Idents(balbc_k7m2_combined.AEC_1),
          balbc_k7m2_combined.AEC_1$sample,
          sep = "_")
balbc_k7m2_combined.AEC_1$celltype <- Idents(balbc_k7m2_combined.AEC_1)
```

#### K7M2 macrophages
```{r k7m2_macs, dependson='subset_epithelial_balbc', eval=TRUE}
Idents(balbc_k7m2_combined) <- "celltype_sample"
###remove "K7M2" from BALBC
balbc_k7m2_combined <-
    subset(balbc_k7m2_combined,
           idents = "K7M2_BALBC",
           invert = T)

###scTransform macs
k7m2_macs <-
    subset(balbc_k7m2_combined,
           idents = "K7M2" &
           cell.type == "PUT STUFF HERE TO GET MACS") %>%
    SCTransform(k7m2_macs,
                vars.to.regress = "percent.mt",
                verbose = FALSE,
                return.only.var.genes = FALSE) %>%
    process_seurat(run_umap_dims = 1:30, resolution = 0.4)

k7m2_mac_markers <-
    FindAllMarkers(k7m2_macs,
                   only.pos = T,
                   logfc.threshold = 0.5,
                   min.pct = 0.25)

###mac cluster annotation based on PMID: 35690521, 32849616, 35365629

###remove contaminating endothelial, T cells
f420_mac_subset <-
    subset(f420_macs,
           idents = c("7", "9"),
           invert = T)

###mac markers based on 35690521 LA-TAM, ANGIO-TAM, IFN-TAM, INFLAM-TAM, PROLIF-TAM, Classical monocyte, nonclassical

# not used
TAM_markers <-
    c("Spp1", "Vegfa", "Fabp5", "Gpnmb", "Lgals3", "Apoe", "Pf4", "Mrc1",
      "Cd206", "Arg1", "Cxcl1", "Cxcl2", "Cxcl3", "Il1a", "Il1b", "Inhba",
      "Ccl20", "Cxcl9", "Cxcl10", "Ifit1", "Ifitm1", "Tnfsf10", "Fn1", "Hmox1",
      "C1qa", "C1qb", "Mki67", "Ace", "Itgal", "Ear2", "Spn", "Itgam", "Itgax",
      "Cd14", "Fcgr3", "Fcgr1", "Cd68", "Cd74", "Cd80", "Cd86", "Ccr2",
      "Cx3cr1", "Ly6c1", "Ly6c2", "H2-Ab1", "H2-DMb1")
###cluster annotation based on PMID: 35690521
```


# Run singler using Brooke's and Ryan's annotations on macrophages
```{r eval=FALSE}
brooke_annot <- qs::qread("~/brooke_annot.qs")

test_annot <-
    SingleR::SingleR(as.SingleCellExperiment(tumor_only),
                         ref = list(GetAssayData(brooke_annot)),
                         labels = list(brooke_annot$cell_typed),
                         aggr.ref = TRUE)
tumor_only$brooke_cell_type <- test_annot$labels
tumor_only$brooke_cell_score <-
    test_annot$scores %>%
    apply(MARGIN = 1, function(x) max(x, na.rm = TRUE))

test_annot <-
    SingleR::SingleR(as.SingleCellExperiment(tumor_only),
                         ref = list(GetAssayData(brooke_annot)),
                         labels = list(brooke_annot$cell_types),
                         aggr.ref = TRUE)
tumor_only$ryan_cell_type <- test_annot$labels
tumor_only$ryan_cell_score <-
    test_annot$scores %>%
    apply(MARGIN = 1, function(x) max(x, na.rm = TRUE))

DimPlot(tumor_only, group.by = c("brooke_cell_type", "ryan_cell_type"))
```
