## Consider Integrating

I'm now going to merge and then annotate the dataset.

```{r looking-for-batch-effects}
merged_samples <- merge(sample_list[[1]], sample_list[-1])

merged_samples <- process_seurat(merged_samples)

DimPlot(merged_samples, group.by = "sample_id")
```

The data looks fairly well integrated. All clusters contain cells from each sample.

### Annotate After Merging

I also want to annotate the merged object. I'm annotating using the MouseRNAseqData and ImmGenData objects from celldex.

```{r annotate-again}
#Going to create a new seurat object
#Join layers first
merged_samples <- annotate_seurat(merged_samples, species = "mouse")

# visualize annotations of new sample
print(DimPlot(merged_samples, group.by = "annotations", label = TRUE, repel = TRUE))
```

## New Reference

I realize that using a single cell dataset as my reference might lead to inappropriate cell type labels, such as adipocytes, astrocytes, hepatocytes, microglia, and neurons. Instead I'm going to use a snRNA-seq dataset as my reference for cell type annotations. The reference dataset is from "Single-Nucleus RNA-Sequencing Profiling of Mouse Lung. Reduced Dissociation Bias and Improved Rare Cell-Type Detection Compared with Single-Cell RNA Sequencing". The GEO accession number is GSE154998, and the metadata was received upon request from the authors.

```{r load-in-new-reference}
sn_mouse_lung <- read.csv("input/GSE145998_Ctrl_mouse_Snuc_comb.dge.txt", sep = ",", header = TRUE) %>%
    column_to_rownames("X")
sn_mouse_lung_md <- read.table("input/GSE145998_nuc_meta.txt")
#meta data doesn't have all the cells that exp matrix has
sn_mouse_lung <- sn_mouse_lung[,rownames(sn_mouse_lung_md)]

mouse_lung_ref <-
    SummarizedExperiment(assays = list(logcounts = log(1 + sn_mouse_lung)),
                         colData = sn_mouse_lung_md)
```

### Re-Run SingleR

```{r sn-singler}
#make merged_samples a SummarizedExperiment
merged_sumexp <- SummarizedExperiment(assays = list(logcounts = log(1 + GetAssayData(merged_samples, slot = "counts"))),
                                      colData = merged_samples@meta.data)

sn_preds <- SingleR(test = merged_sumexp,
                    ref = mouse_lung_ref,
                    labels = mouse_lung_ref$cell_type)
```

### Add Predictions to Seurat Object

Now I'll add these predictions back to my seurat object and visualize them.

I'm also going to add a column called "coarse_cell_type" that has more general annotations based on the single nuclei cell type predictions.

```{r add-sn-preds}
merged_samples <- AddMetaData(object = merged_samples,
                              metadata = sn_preds$labels,
                              col.name = "sn_cell_type")

DimPlot(merged_samples, group.by = "sn_cell_type", label = TRUE)

# Make coarser labels
merged_samples@meta.data <-
    mutate(merged_samples@meta.data,
           coarse_cell_type = recode(sn_cell_type,
                                     "AT1" = "Epithelial",
                                     "AT2" = "Epithelial",
                                     "Epi" = "Epithelial",
                                     "aEC" = "Endothelial",
                                     "CapEC" = "Endothelial",
                                     "EC1" = "Endothelial",
                                     "LEC" = "Endothelial",
                                     "vEC" = "Endothelial",
                                     "AM" = "Mac",
                                     "IM" = "Mac",
                                     "BC" = "B cell",
                                     "GB" = "B cell",
                                     "FB1" = "FB",
                                     "FB2" = "FB",
                                     "matFB" = "FB",
                                     "MyoFB" = "FB",
                                     "TC" = "T cell",
                                     "TC3" = "T cell",
                                     "Th17 TC" = "T cell"))

DimPlot(merged_samples, group.by = "coarse_cell_type", label = TRUE)
```

### Cell Type Counts

```{r cell-type-count-table}
merged_samples@meta.data %>%
    select(sample_id, coarse_cell_type) %>%
    group_by(sample_id, coarse_cell_type) %>%
    mutate(total = n()) %>%
    unique() %>%
    pivot_wider(names_from = "sample_id", values_from = "total") 
```

What the cell type abbreviations are

- aEC: arterial endothelial cells
- AM: alveolar macs
- AT1: alveolar type 1 epithelial cells
- AT2: alveolar type 2 epithelial cells
- BC: B cells
- CapEC: capillary endothelial cells
- DC: dendritic cells
- Div: dividing cells
- EC: endothelial cells
- Epi: epithelial cells
- FB: fibroblasts
- GB: germinal B cells
- IM: interstitial macs
- LEC: lymphatic endothelial cells
- Mes: mesolthelial cells
- Mono: monocytes
- MyoFB: myofibroblasts
- Peri: pericytes
- SMC: smooth muscle cells
- TC: T cells
